.github/workflows/merge-to-stage.js

```diff
@@ -211,11 +211,45 @@ const openStageToMainPR = async () => {
 
 const mergeLimitExceeded = () => MAX_MERGES - existingPRCount < 0;
 
+const workingHours = () => {
+  const now = new Date();
+  const day = now.getUTCDay();
+  const isFriday = day === 5;
+  const isSaturday = day === 6;
+  const isSunday = day === 0;
+  return !isFriday && !isSaturday && !isSunday;
+};
+
+const thursdayBeforeRCP = () => {
+  // This function is used to block creating stage batches on Thursdays (4) before an RCP on monday next week
+  const now = new Date();
+  const day = now.getUTCDay();
+  
+  if (day !== 4) return false;
+  
+  const nextMonday = new Date(now);
+  nextMonday.setUTCDate(now.getUTCDate() + 4);
+  
+  const { RCPDates } = require('./helpers.js');
+  
+  return RCPDates.some(({ start }) => {
+    const rcpStart = new Date(start);
+    return rcpStart.toDateString() === nextMonday.toDateString();
+  });
+};
+
 const main = async (params) => {
   github = params.github;
   owner = params.context.repo.owner;
   repo = params.context.repo.repo;
-  if (isWithinRCP({ offset: process.env.STAGE_RCP_OFFSET_DAYS || 2, excludeShortRCP: true })) return console.log('Stopped, within RCP period.');
+  
+  if (isWithinRCP({ offset: process.env.STAGE_RCP_OFFSET_DAYS || 2, excludeShortRCP: true }) || thursdayBeforeRCP()) {
+    return console.log('Stopped, within RCP period.');
+  }
+  
+  if (!workingHours()) {
```

Comment:
Not sure about the workingHours.. usually if we merge something manually on a friday, it's wanted to create a new batch immediately so it can be tested and we are faster ready for the next merge on monday